src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaService.java

```diff
@@ -27,4 +27,17 @@
     List<SagaLog> getLogsByTransaction(String txId);
 
     Page<SagaTransaction> getAllTransaction(Pageable pageable);
+
+    /**
+     * This method restore activation process in a low probability сase when kafka lost events.
+     * For example after disk problem, like all disk was full etc.
+     */
+    void resendAllEventsInQueue();
```

Comment:
suggest to rename `resendAllEventsInQueue` to `resendEventsByStateInQueue` to avoid `All` and narrowing definition `InQueue` method.

(or create method with parameter `resendEventsByState(state)`)